### PR TITLE
Update YamlDotNet; add test for yaml

### PR DESCRIPTION
**Pull request by tobindh**
Update YamlDotNet to latest version, per issue #14 - also add a simple test for the YamlProvider.

### CODE REVIEW

**Comment by tobindh** [comment]
@laura-rodriguez @nbarbettini Added PR, whenever you have time to approve - this would allow upgrading the Okta SDK and prevent issues with other projects using later versions of yamldotnet.

File: src/FlexibleConfiguration/Providers/YamlVisitors/ContextAwareVisitor.cs [hunk 1]
```diff
@@ -90,9 +90,12 @@ private static bool IsNull(YamlScalarNode scalar)
                 return true;
             }
 
-            if (string.Equals(scalar.Tag, "tag:yaml.org,2002:null", StringComparison.OrdinalIgnoreCase))
+            if (!scalar.Tag.IsEmpty)
```
Comment (laura-rodriguez) [inline src/FlexibleConfiguration/Providers/YamlVisitors/ContextAwareVisitor.cs]: Nit: Can you please combine the nested ifs in one condition?

```
if(!scalar.Tag.IsEmpty && string.Equals(scalar.Tag.Value, "tag:yaml.org,2002:null", StringComparison.OrdinalIgnoreCase)
```

**Review by laura-rodriguez — APPROVED** [review]
Minor nit-pick, but other than that LGTM 👍 .
Tested locally ✔️ 

cc @nbarbettini

**Review by nbarbettini — APPROVED** [review]

**Comment by nbarbettini** [comment]
Thanks @tobindh! I am going to further bump `YamlDotNet` from 11.1.1 to 11.2.1 (the latest at the time of this writing).

**Comment by nbarbettini** [comment]
This has been published on Nuget -- see my comments on https://github.com/nbarbettini/FlexibleConfiguration/issues/14.

**Pull request merged**